src/Room.java

```diff
@@ -26,22 +26,26 @@ void setCommoditySet(HashSet<AbstractCommodity> givenCommoditySet){
         this.commoditySet = givenCommoditySet;
     }
 
-    //Prepares commodities in a room & adds a maintenance record through string
-    void prepareCommodities(String date) throws ParseException {
-        //Iterator<AbstractCommodity> itr = this.commoditySet.iterator();
-        for (AbstractCommodity s : this.commoditySet) {
-            s.prepare();
+    Date stringToDate(String stringDate){
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy/mm/dd");
+        try {
+
+            Date date = formatter.parse(stringDate);
+            return date;
+
+        } catch (ParseException e) {
+            System.out.println("Parse throws exception");
+            return null;
         }
-        Date date1 = new SimpleDateFormat("yyyy/mm/dd").parse(date);
-        this.maintenanceDates.add(date1);
+
     }
 
-    //Prepares commodities in a room & adds a maintenance record with current date
-    void prepareCommoditiesWithCurrentTime(){
+    //Prepares commodities in a room & adds a maintenance record through string
+    void prepareCommodities(Date date) {
```

Comment:
Also null checks. In here you don't really know if you receive a valid date, it can be null.
The commodity set may not be instantiated, which will result in a nullPointerException.